src/core/reducers/abuse.js

```diff
@@ -4,53 +4,85 @@ import invariant from 'invariant';
 import type { AddonType } from 'core/types/addons';
 import type { AbuseReporter } from 'core/api/abuse';
 
-export const DISABLE_ADDON_ABUSE_BUTTON_UI = 'DISABLE_ADDON_ABUSE_BUTTON_UI';
-export const ENABLE_ADDON_ABUSE_BUTTON_UI = 'ENABLE_ADDON_ABUSE_BUTTON_UI';
-export const HIDE_ADDON_ABUSE_REPORT_UI = 'HIDE_ADDON_ABUSE_REPORT_UI';
+export const DISABLE_ADDON_ABUSE_BUTTON_UI: 'DISABLE_ADDON_ABUSE_BUTTON_UI' =
+  'DISABLE_ADDON_ABUSE_BUTTON_UI';
+export const ENABLE_ADDON_ABUSE_BUTTON_UI: 'ENABLE_ADDON_ABUSE_BUTTON_UI' =
+  'ENABLE_ADDON_ABUSE_BUTTON_UI';
+export const HIDE_ADDON_ABUSE_REPORT_UI: 'HIDE_ADDON_ABUSE_REPORT_UI' =
+  'HIDE_ADDON_ABUSE_REPORT_UI';
 export const LOAD_ADDON_ABUSE_REPORT: 'LOAD_ADDON_ABUSE_REPORT' =
   'LOAD_ADDON_ABUSE_REPORT';
 export const SEND_ADDON_ABUSE_REPORT: 'SEND_ADDON_ABUSE_REPORT' =
   'SEND_ADDON_ABUSE_REPORT';
-export const SHOW_ADDON_ABUSE_REPORT_UI = 'SHOW_ADDON_ABUSE_REPORT_UI';
+export const SHOW_ADDON_ABUSE_REPORT_UI: 'SHOW_ADDON_ABUSE_REPORT_UI' =
+  'SHOW_ADDON_ABUSE_REPORT_UI';
 
-type DisableAddonAbuseButtonUIType = {| addon: AddonType |};
+export type AddonAbuseState = {|
+  buttonEnabled?: boolean,
+  message: string,
+  reporter: AbuseReporter | null,
+  uiVisible?: boolean,
+|};
+
+export type AbuseState = {|
+  bySlug: {
+    [addonSlug: string]: AddonAbuseState,
+  },
+  loading: boolean,
+|};
+
+export const initialState = {
```

Comment:
I've looked at some other files and I see they show the type here too:

`export const initialState : AbuseState`

Reply:
yep